5주차/5107/5107_후디_python.py

```diff
@@ -0,0 +1,40 @@
+def dfs(graph, node): # 순환여부를 boolean으로 반환
+    global visited
+
+    visited.append(node)
+    if node not in graph.keys(): # 이 사람은 선행을 베풀지 않음
+        return False
+
+    neighbor = graph[node]
+    if neighbor in visited: # 순환 발견
+        return True
+
+    return dfs(graph, neighbor)
+
+cases = []
+while True:
+    N = int(input())
+    if N == 0:
+        break
+
```

Comment:
```suggestion
while (N:=int(input())) != 0:

```
19년도 python 3.8부터는 `:=` 바다코끼리 연산자가 추가되어, 위와 같이 사용할 수 있습니다~!!
https://docs.python.org/ko/3/whatsnew/3.8.html

Reply:
오호 신기한 연산자네요! 감사합니다 이프!